homeassistant/components/verisure/coordinator.py

```diff
@@ -56,60 +57,104 @@ async def async_login(self) -> bool:
 
     async def async_logout(self, _event: Event) -> None:
         """Logout from Verisure."""
-        try:
-            await self.hass.async_add_executor_job(self.verisure.logout)
-        except VerisureError as ex:
-            LOGGER.error("Could not log out from verisure, %s", ex)
 
     async def _async_update_data(self) -> dict:
         """Fetch data from Verisure."""
+        try:
+            await self.hass.async_add_executor_job(self.verisure.update_cookie)
+        except (VerisureLoginError, VerisureResponseError) as ex:
```

Comment:
What do I need to raise `UpdateFailed` in case of re-auth or just log a debug message?

Reply:
I'm thinking of adding the overview object to the coordinator so that it returns the old overview in case of a re-auth (cookie expired).

Reply:
What is the best approach the handle a re-auth (update_cookie)?
If I raise `UpdateFailed`, I'm raising an error for an expected course of events.
If I return None, all the entities become unavailabe.
If I maintain a copy of the prior overview, I'm adding all this unneccecery data to the coordinator.

Reply:
I guess I could just `return await self._async_update_data()`

Reply:
I was thinking of doing it like this but then we just end up with an error in the logs every 15 minutes.

```py
        try:
            overview = await self.hass.async_add_executor_job(
                self.verisure.request,
                self.verisure.arm_state(),
                self.verisure.broadband(),
                self.verisure.cameras(),
                self.verisure.climate(),
                self.verisure.door_window(),
                self.verisure.smart_lock(),
                self.verisure.smartplugs(),
            )
        except VerisureResponseError as err:
            try:
                await self.hass.async_add_executor_job(self.verisure.update_cookie)
            except VerisureResponseError as ex:
                LOGGER.error("Credentials expired for Verisure, %s", ex)
                raise ConfigEntryAuthFailed("Credentials expired for Verisure") from ex
            except Exception as ex:
                LOGGER.error("Could not read overview, %s", ex)
                raise UpdateFailed("Could not read overview") from ex
            raise UpdateFailed("Cookies expired") from err
        except VerisureError as ex:
            LOGGER.error("Could not read overview, %s", ex)
            raise UpdateFailed("Could not read overview") from ex
```

Instead of `raise UpdateFailed("Cookies expired") from err`, I could `return await self._async_update_data()`, but there is a risk of an infinite loop.

Reply:
Any suggestions?

Reply:
Why is there not an `UpdateDelayed` exception that passes silently?

Reply:
resolved it by adding a `self._overview` to the coordinator and saving the last working overview to it.